### PR TITLE
Delta ll individual

### PR DESCRIPTION
**Pull request by peterstangl**
This PR adds an individual log likelihood to the obstable and introduces some additional options in the obstable method for sorting the table.

### CODE REVIEW

File: smelli/classes.py [hunk 1]
```diff
@@ -135,6 +137,58 @@ def log_likelihood_sm(self):
...
+                    info[obs]['lh_name'] = flh_name
+                    info[obs]['name'] = obs if isinstance(obs, str) else obs[0]
+                    info[obs]['theory_sm'] = pred_sm[obs]
+                    info[obs]['th. unc.'] = np.sqrt(sm_cov[i, i])
+                    info[obs]['experiment'] = m.get_central(obs)
+                    info[obs]['exp. unc.'] = np.sqrt(exp_cov[i, i])
+                    info[obs]['exp. PDF'] = NormalDistribution(m.get_central(obs), np.sqrt(exp_cov[i, i]))
+                    info[obs]['inspire'] = sorted(set(inspire_dict[obs]))
+                    info[obs]['ll_sm'] = m.get_logprobability_single(obs, pred_sm[obs])
+                    info[obs]['ll_central'] = m.get_logprobability_single(obs, m.get_central(obs))
```
Comment (DavidMStraub) [inline smelli/classes.py:162]: Here the column names are not uniform. The "old" ones are abbreviations, the new ones written like variables. So I would replace `theory_sm` simply by `SM`. `ll_sm`, `ll_central` and `lh_name` are deleted later, so it doesn't really matter.

File: smelli/classes.py [hunk 2]
```diff
@@ -229,82 +295,41 @@ def log_likelihood_global(self):
...
+                    ll = p_comb.logpdf([pred[obs]])
+                    info[obs]['pull'] = pull(-2 * (ll - ll_central), dof=1)
+                    info[obs]['delta_ll'] = (ll - ll_sm)[0]
             self._obstable_tree_cache = info
         return self._obstable_tree_cache
 
-    def obstable(self, min_pull=0):
+    def obstable(self, min_pull=0, sort_by='pull', min_val=None, max_val=None,
+                 ascending = None):
         r"""Return a pandas data frame with the central values and uncertainties
```
Comment (DavidMStraub) [inline smelli/classes.py:333]: no spaces around = :)

File: smelli/classes.py [hunk 3]
```diff
@@ -313,28 +338,51 @@ def obstable(self, min_pull=0):
         due to correlations.
         """
         info = self._obstable_tree
-        info = self._obstable_filter_sort(info, min_pull=min_pull,
-                                          sortkey='pull', reverse=True)
+        subset = None
+        if sort_by == 'pull':
+            if ascending is None:
+                ascending = False
```
Comment (DavidMStraub) [inline smelli/classes.py:353]: What's that? Why is the default value not `False` then?

Comment (DavidMStraub) [reply smelli/classes.py:353]: Oh sorry, now I get it. It depends on what it is sorted by, OK.
Could you please add a few comments?

File: smelli/classes.py [hunk 4]
```diff
@@ -313,28 +338,51 @@ def obstable(self, min_pull=0):
...
+                                          min_val=min_val, max_val=max_val,
+                                          subset=subset)
         # create DataFrame
         df = pd.DataFrame(info).T
         # remove inspire references, likelihood name, and observable name
-        del(df['inspire'])
-        del(df['lh_name'])
-        del(df['name'])
-        del(df['exp. PDF'])
+        if len(df) >0:
```
Comment (DavidMStraub) [inline smelli/classes.py:371]: When is the length 0? Can you add a comment please?

File: smelli/classes.py (hunk 1 above)
Comment (peterstangl) [reply smelli/classes.py:162]: At the moment, `theory_sm` is also deleted later. Do you think it should be included in the obstable output?

Comment (DavidMStraub) [reply smelli/classes.py:162]: Oh. Well it's not strictly necessary, I thought it was intentional. Can also be omitted.

Comment (peterstangl) [reply smelli/classes.py:162]: I just realized that `theory_sm` it is not used at all later on. So I will just completely remove it.

Comment (peterstangl) [reply smelli/classes.py:162]: Done in https://github.com/smelli/smelli/pull/1/commits/3a768107a4709b27a728ce160e32eb3e7b794920.

File: smelli/classes.py (hunk 2 above)
Comment (peterstangl) [reply smelli/classes.py:333]: Done in https://github.com/smelli/smelli/pull/1/commits/c44bac3e253812c62a818a0b52bc80c9a56a402f.

File: smelli/classes.py (hunk 3 above)
Comment (peterstangl) [reply smelli/classes.py:353]: Done in https://github.com/smelli/smelli/pull/1/commits/3f968936b1d471f26c63e697294e53110a6a4632.

File: smelli/classes.py (hunk 4 above)
Comment (peterstangl) [reply smelli/classes.py:371]: Done in https://github.com/smelli/smelli/pull/1/commits/3f968936b1d471f26c63e697294e53110a6a4632.

**Comment by peterstangl** [comment]
@DavidMStraub can this be merged?

**Comment by DavidMStraub** [comment]
Oh, sure. Thanks

**Pull request merged**